### PR TITLE
Step 4: kernel integration (fvm implements VirtualMachine)

### PR DESCRIPTION
**Pull request by aionick**
Step 4 in the integration.

This PR is mostly concerned with getting the fast vm to implement the `VirtualMachine` interface defined in the VM API as well as getting the `BulkExecutor` to start using this new implementing class.

See the corresponding PR's up in the sub-repositories here:
- [FastVM](https://github.com/aionnetwork/aion_fastvm/pull/45)
- [Aion API](https://github.com/aionnetwork/aion_api/pull/54)

### CODE REVIEW

File: modMcf/test/org/aion/mcf/trie/TrieTestWithRootHashValues.java [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+
+package org.aion.mcf.trie;
+>>>>>>> c7a85d2... separating insert and delete functionality inside the trie implementation:modMcf/test/org/aion/mcf/trie/TrieTestWithRootHashValues.java
```
Comment (AionJayT) [inline modMcf/test/org/aion/mcf/trie/TrieTestWithRootHashValues.java]: merge conflict

File: modMcf/test/org/aion/mcf/trie/TrieTest.java [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+
+package org.aion.mcf.trie;
+>>>>>>> c7a85d2... separating insert and delete functionality inside the trie implementation:modMcf/test/org/aion/mcf/trie/TrieTest.java
```
Comment (AionJayT) [inline modMcf/test/org/aion/mcf/trie/TrieTest.java]: merge conflict

**Comment by AlexandraRoatis** [comment]
this PR is work in progress, not ready for review

File: modAionImpl/test/org/aion/zero/impl/vm/ContractIntegTest.java [hunk 3]
```diff
@@ -808,50 +830,52 @@ public void testRecursiveStackoverflow() throws IOException {
...
-//        assertEquals(FastVmResultCode.SUCCESS, result.getResultCode());
-//        assertEquals(nrg - tx.getNrgConsume(), result.getEnergyRemaining());
-//        assertNotEquals(nrg, tx.getNrgConsume());
-//
-//        // Check that we actually did call the contract and received its output.
-//        String expectedMsg = CallMePrecompiledContract.head + tagToSend;
-//        byte[] output = result.getOutput();
-//        assertArrayEquals(expectedMsg.getBytes(), output);
-//        assertTrue(CallMePrecompiledContract.youCalledMe);
+        //        String tagToSend = "Soo cool!";
```
Comment (aion-kelvin) [inline modAionImpl/test/org/aion/zero/impl/vm/ContractIntegTest.java]: Are we planning to re-enable these at some point before AVM is released?

**Pull request merged**